dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/WorkflowItemRestRepository.java

```diff
@@ -0,0 +1,335 @@
+        WorkflowItemRest wsi = findOne(id);
+        XmlWorkflowItem source = wis.find(context, id);
+        for (Operation op : operations) {
+            //the value in the position 0 is a null value
+            String[] path = op.getPath().substring(1).split("/", 3);
+            if (OPERATION_PATH_SECTIONS.equals(path[0])) {
+                String section = path[1];
+                evaluatePatch(context, request, source, wsi, section, op);
+            } else {
+                throw new PatchBadRequestException(
+                    "Patch path operation need to starts with '" + OPERATION_PATH_SECTIONS + "'");
+            }
+        }
+        wis.update(context, source);
+    }
+
+    private void evaluatePatch(Context context, HttpServletRequest request, XmlWorkflowItem source,
+                               WorkflowItemRest wsi, String section, Operation op) {
+        SubmissionConfig submissionConfig =
+            submissionConfigReader.getSubmissionConfigByName(wsi.getSubmissionDefinition().getName());
+        for (int stepNum = 0; stepNum < submissionConfig.getNumberOfSteps(); stepNum++) {
+
+            SubmissionStepConfig stepConfig = submissionConfig.getStep(stepNum);
+
+            if (section.equals(stepConfig.getId())) {
+                /*
+                 * First, load the step processing class (using the current
+                 * class loader)
+                 */
+                ClassLoader loader = this.getClass().getClassLoader();
+                Class stepClass;
+                try {
+                    stepClass = loader.loadClass(stepConfig.getProcessingClassName());
+
+                    Object stepInstance = stepClass.newInstance();
+
+                    if (stepInstance instanceof AbstractRestProcessingStep) {
+                        // load the JSPStep interface for this step
+                        AbstractRestProcessingStep stepProcessing =
+                            (AbstractRestProcessingStep) stepClass.newInstance();
+                        stepProcessing.doPreProcessing(context, source);
+                        stepProcessing.doPatchProcessing(context, getRequestService().getCurrentRequest(), source, op);
+                        stepProcessing.doPostProcessing(context, source);
+                    } else {
+                        throw new PatchBadRequestException(
+                            "The submission step class specified by '" + stepConfig.getProcessingClassName() +
+                            "' does not extend the class org.dspace.submit.AbstractProcessingStep!" +
+                            " Therefore it cannot be used by the Configurable Submission as the <processing-class>!");
+                    }
+
+                } catch (Exception e) {
+                    log.error(e.getMessage(), e);
+                }
+            }
+        }
+    }
+
+    @Override
+    protected void delete(Context context, Integer id) {
```

Comment:
Is there a need for a separate delete function here?
The configurable workflow handles this as the result of an action

Reply:
Actually, I'm not happy with the solution that we have had to implement to support the configurable actions. I find that the current solution described here
https://github.com/DSpace/Rest7Contract/blob/master/claimedtasks.md#post-method-single-resource-level

is not really restful and not at all HATEOAS. To change it we need to discuss a better contract and make some changes to the underlying XmlWorkflow framework that I preferred to postponed to don't increase the size of this PR.

That said, the best that we were able to do were actually identify a restful way to deal with the unclaim use case and it was documented in the initial contract
https://github.com/DSpace/Rest7Contract/blob/master/claimedtasks.md#delete-method

Reply:
sorry, the previous comment is not really related to your question. It is focus on ClaimedTaskRestRepository#delete (your next comment) instead than WorkflowItemRestRepository#delete

The WorkflowItemRestRepository#delete provides support for the "Abort workflow" functionality that is supported out of box in the Basic Workflow and JSPUI (not sure about XMLUI), It was documented here
https://github.com/DSpace/Rest7Contract/blob/master/workflowitems.md#delete-method

Reply:
Abort a workflow and the reject action are two different things. Abort is a special function reserved to administrator, it is described here: https://wiki.duraspace.org/display/DSDOC5x/Functional+Overview#FunctionalOverview-SubmissionWorkflowinDSpace

the reject action is (depending on the reach state) available to reviewers and in the standard workflow it produces the same effect than an abort but of course with the configurable workflow it could lead also to other effects (as for example back to a previous workflow step instead than to the submitter)

Reply:
Thanks @abollini for pointing me at the difference in specifications.
I didn't realize this was a feature implemented in the configurable workflow because it's not made available anywhere in DSpace 6 XMLUI.

Since this action is quite similar to the reject action (both send the item to the submitter), but this is an admin-only feature which is available for all steps, it would be best to add some JavaDocs to this method

Reply:
done